deployment/common/changeset/example/link_transfer.go

```diff
@@ -0,0 +1,182 @@
+		if err != nil {
+			return fmt.Errorf("invalid chain selector: %w", err)
+		}
+		chain := e.Chains[chainSel]
+		addrs, err := e.ExistingAddresses.AddressesForChain(chainSel)
+		if len(transfers) == 0 {
+			return fmt.Errorf("transfers for chainSel %d must have at least one LinkTransfer", chainSel)
+		}
+		totalAmount := big.NewInt(0)
+		linkState, err := changeset.MaybeLoadLinkTokenState(chain, addrs)
+		if err != nil {
+			return fmt.Errorf("error loading link token state during validation: %w", err)
+		}
+		for _, transfer := range transfers {
+			if transfer.To == (common.Address{}) {
+				return fmt.Errorf("'to' address for transfers  must be set")
+			}
+			if transfer.Value == nil {
+				return fmt.Errorf("value for transfers must be set")
+			}
+			if transfer.Value == big.NewInt(0) {
+				return fmt.Errorf("value for transfers must be non-zero")
+			}
+		}
+		// check that from address has enough funds for the transfers
+		balance, err := linkState.LinkToken.BalanceOf(&bind.CallOpts{Context: ctx}, cfg.From)
+		if balance.Cmp(totalAmount) < 0 {
+			return fmt.Errorf("timelock address does not have enough funds for transfers for chain selector %d", chainSel)
+		}
+	}
+
+	if cfg.McmsConfig == nil {
+		return nil
+	}
+
+	// Upper bound for min delay (7 days)
+	if cfg.McmsConfig.MinDelay > MaxTimelockDelay {
+		return fmt.Errorf("minDelay must be less than 7 days")
+	}
+	return nil
+}
+
+// LinkTransfer takes the given link transfers and executes them or creates an MCMS proposal for them.
+func LinkTransfer(e deployment.Environment, cfg *LinkTransferConfig) (deployment.ChangesetOutput, error) {
+
+	err := cfg.Validate(e)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("invalid LinkTransferConfig: %w", err)
+	}
+	chainSelectors := []uint64{}
+	for chainSelector := range cfg.Transfers {
+		chainSelectors = append(chainSelectors, chainSelector)
+	}
+	mcmsPerChain := map[uint64]*owner_helpers.ManyChainMultiSig{}
+
+	timelockAddresses := map[mcms.ChainIdentifier]common.Address{}
+	allBatches := []timelock.BatchChainOperation{}
+	for chainSelector := range cfg.Transfers {
+		chainID := mcms.ChainIdentifier(chainSelector)
+		chain := e.Chains[chainSelector]
+		addrs, err := e.ExistingAddresses.AddressesForChain(chainSelector)
+		linkState, err := changeset.MaybeLoadLinkTokenState(chain, addrs)
+		if err != nil {
+			return deployment.ChangesetOutput{}, err
+		}
+		linkAddress := linkState.LinkToken.Address()
+		mcmsState, err := changeset.MaybeLoadMCMSWithTimelockState(chain, addrs)
+		if err != nil {
+			return deployment.ChangesetOutput{}, err
+		}
+		timelockAddress := mcmsState.Timelock.Address()
+
+		mcmsPerChain[uint64(chainID)] = mcmsState.ProposerMcm
+
+		timelockAddresses[chainID] = timelockAddress
+		batch := timelock.BatchChainOperation{
+			ChainIdentifier: chainID,
+			Batch:           []mcms.Operation{},
+		}
+		opts := chain.DeployerKey
+		if cfg.McmsConfig != nil {
```

Comment:
Maybe we should follow the [getDeployer pattern](https://github.com/smartcontractkit/chainlink/blob/88a0338576dad34068f8ee278a5cf5ac0b8870ad/deployment/ccip/changeset/cs_update_rmn_config.go#L21-L26)?

Reply:
added https://github.com/smartcontractkit/chainlink/pull/15512/commits/9bc7467f52565089b5e1a420de6fe0f88be42ccf